AVFoundation-CombineTests/PlayheadProgressPublisherTests.swift

```diff
@@ -0,0 +1,151 @@
+        }
+        
+        XCTAssertEqual(receivedTimes, expectedTimes)
+    }
+    
+    func testWhenNoValuesAreRequested_ItEmitsNoValues() {
+        // given
+        let expectedValues: [TimeInterval] = []
+        var receivedValues: [TimeInterval] = []
+        
+        let subscriber = TestSubscriber<TimeInterval>(demand: 0) { values in
+            receivedValues = values
+        }
+        
+        sut.subscribe(subscriber)
+        
+        let timeUpdates: [TimeInterval] = [1, 2, 3, 4, 5]
+        
+        // when
+        timeUpdates.forEach { time in player.updateClosure?(CMTime(seconds: time, preferredTimescale: CMTimeScale(NSEC_PER_SEC))) }
+        
+        // then
+        XCTAssertEqual(receivedValues, expectedValues)
+    }
+    
+    func testWhenValuesAreRequested_ItStartsEmittingValues() {
+        // given
+        let expectedValues: [TimeInterval] = [1, 2, 3, 4, 5]
+        var receivedValues: [TimeInterval] = []
+        
+        let subscriber = TestSubscriber<TimeInterval>(demand: 0) { values in
+            receivedValues = values
+        }
+        
+        sut.subscribe(subscriber)
+        
+        let timeUpdates: [TimeInterval] = [1, 2, 3, 4, 5]
+        
+        // when
+        subscriber.startRequestingValues(5)
+        
+        timeUpdates.forEach { time in player.updateClosure?(CMTime(seconds: time, preferredTimescale: CMTimeScale(NSEC_PER_SEC))) }
+        
+        // then
+        XCTAssertEqual(receivedValues, expectedValues)
+    }
+    
+    func testWhenOnlyOneValueIsRequested_ItCompletesAfterEmittingOneValue() {
+        // given
+        let expectedValues: [TimeInterval] = [1]
+        var receivedValues: [TimeInterval] = []
+        
+        let subscriber = TestSubscriber<TimeInterval>(demand: 1) { values in
+            receivedValues = values
+        }
+        
+        sut.subscribe(subscriber)
+        
+        let timeUpdates: [TimeInterval] = [1, 2, 3, 4, 5]
+        
+        // when
+        timeUpdates.forEach { time in player.updateClosure?(CMTime(seconds: time, preferredTimescale: CMTimeScale(NSEC_PER_SEC))) }
+        
+        // then
+        XCTAssertEqual(receivedValues, expectedValues)
+    }
+    
+    func testWhenTwoValuesAreRequested_ItCompletesAfterEmittingTwoValues() {
+        // given
+        let expectedValues: [TimeInterval] = [1, 2]
+        var receivedValues: [TimeInterval] = []
+        
+        let subscriber = TestSubscriber<TimeInterval>(demand: 2) { values in
+            receivedValues = values
+        }
+        
+        sut.subscribe(subscriber)
+        
+        let timeUpdates: [TimeInterval] = [1, 2, 3, 4, 5]
+        
+        // when
+        timeUpdates.forEach { time in player.updateClosure?(CMTime(seconds: time, preferredTimescale: CMTimeScale(NSEC_PER_SEC))) }
+        
+        // then
+        XCTAssertEqual(receivedValues, expectedValues)
+    }
+}
+
```

Comment:
```suggestion

/// Mock implementation of `AVPlayer`.
/// This particular sub-class overrides `addPeriodicTimeObserver(forInterval:queue:using)` so that we can store a reference to the last parameter it is passed and in this way "fake" the `AVPlayer`'s playhead moving without actually having to play any media.
```